### DISCUSSION TITLE
Alpine.js now missing from TALL stack? Am I missing something

### DISCUSSION THREAD

**Discussion by teaforchris** [laravel-frontend-presets/tall#156] [General]
This commit https://github.com/laravel-frontend-presets/tall/commit/aef809373eac9f9c27348716d24eb9b7c9c48314 means that any newly created project will not have AlpineJS out of the box - so I manually needed to install and put this back.

I don't quite understand this commit! Am I missing something?

**Comment by ryangjchandler** [comment]
@teaforchris Livewire ships with Alpine.js out of the box, since it is built on top of Livewire.

**Comment by teaforchris** [comment]
Ok so why doesn't this work?

```
<button type="button" class="-m-2.5 p-2.5 text-gray-700 lg:hidden" x-on:click="alert('set sidebar open');">
        <span class="sr-only">Open sidebar</span>
        <x-heroicon-o-bars-3 class="h-6 w-6" aria-hidden="true" />
    </button>
```

Expect to see an alert when I click the button.

This is just in a regular blade template.

I'm completely new to Livewire, sorry.

**Reply by danharrin** [reply]
Because 1) it isn't in a Livewire component or 2) you havent initialized an Alpine component with `x-data="{}"`

**Reply by teaforchris** [reply]
Interesting. So I can't just use JS directly in a regular blade file - it _has_ to be a livewire component?

**Reply by imliam** [reply]
You should be able to use it outside Livewire components - have you initialised a component with `x-data` like Dan mentioned?

**Reply by danharrin** [reply]
Alpine OR Livewire component
